src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -717,28 +717,34 @@ private PropertyInfo PropertyInfo
             /// <returns>Syntax for retrieving the value of this field.</returns>
             public ExpressionSyntax GetGetter(ExpressionSyntax instance, bool forceAvoidCopy = false)
             {
-                var typeSyntax = this.FieldInfo.FieldType.GetTypeSyntax();
-                var getFieldExpression =
-                    SF.InvocationExpression(SF.IdentifierName(this.GetterFieldName))
-                        .AddArgumentListArguments(SF.Argument(instance));
+                Type type;
+                ExpressionSyntax getExpression;
 
                 // If the field is the backing field for an auto-property, try to use the property directly.
                 if (this.PropertyInfo != null && this.PropertyInfo.GetGetMethod() != null)
                 {
-                    return instance.Member(this.PropertyInfo.Name);
+                    type = this.PropertyInfo.PropertyType;
+                    getExpression = instance.Member(this.PropertyInfo.Name);
+                }
+                // otherwise, construct a field access expression
+                else
+                {
+                    type = this.FieldInfo.FieldType;
+                    getExpression = SF.InvocationExpression(SF.IdentifierName(this.GetterFieldName))
```

Comment:
Good catch - I see that I was returning the property expression instead of checking if it needed to be deep copied first.
This will conflict with #1241, so hopefully we remember to fix it.
`type` here surely cannot vary between auto-property and the backing field.